### DISCUSSION TITLE
How to try the quic implementation

### DISCUSSION THREAD

**Discussion by ackintosh** [libp2p/rust-libp2p#4055] [Q&A] [answered]
Hi! I'm trying to test the quic implementation that is in progress on https://github.com/libp2p/rust-libp2p/pull/3454 (using Testground).

[Here](https://github.com/ackintosh/gossipsub-testground/pull/3) is a test I'm working on however a build error below occurs:

https://github.com/ackintosh/gossipsub-testground/blob/07da4371c886f73e20c230e1087d2e8ab1fbd3a4/flood_publishing/src/network.rs#L86-L91

```rust
        let swarm = SwarmBuilder::with_tokio_executor(
            libp2p_quic::tokio::Transport::new(libp2p_quic::Config::new(&keypair)).boxed(),
            gossipsub,
            PeerId::from(keypair.public()),
        )
```

```
error[E0308]: mismatched types
    --> src/network.rs:87:13
     |
86   |         let swarm = SwarmBuilder::with_tokio_executor(
     |                     --------------------------------- arguments to this function are incorrect
87   |             libp2p_quic::tokio::Transport::new(libp2p_quic::Config::new(&keypair)).boxed(),
     |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected `Boxed<(PeerId, StreamMuxerBox)>`, found `Boxed<(PeerId, Connection)>`
     |
     = note: expected struct `Boxed<(PeerId, StreamMuxerBox)>`
                found struct `Boxed<(PeerId, libp2p_quic::Connection)>`
```

I couldn't find a builder function that takes `Boxed<(PeerId, libp2p_quic::Connection)>`.  I might be doing something wrong. I would appreciate it if you could tell me the correct way to do it. 🙏

**Comment by mxinden** [comment, marked as answer]
Seems like you are not wrapping the `libp2p_quic::Connection` in a `StreamMuxerBox`. Is the example below of some help?

https://github.com/mxinden/rust-libp2p/blob/84d29b34b19aef161a8583daffb9f63a385a613b/protocols/perf/src/bin/perf.rs#L395-L425

**Reply by ackintosh** [reply]
That is so helpful. Thank you!
